swagger/operation.go

```diff
@@ -7,13 +7,30 @@ import (
 	"github.com/go-openapi/spec"
 )
 
+// SingleInputSchemaParameterOp returns true if the operation has a single,
+// schema'd body input. It also returns the name of the model (without "models.").
+func SingleSchemaedBodyParameter(op *spec.Operation) (bool, string) {
+	if len(op.Parameters) == 1 && op.Parameters[0].ParamProps.Schema != nil {
+		typ, _ := TypeFromSchema(op.Parameters[0].ParamProps.Schema, false)
+		return true, typ
+	}
+	return false, ""
+}
+
 // Interface returns the interface for an operation
 func Interface(op *spec.Operation) string {
 	capOpID := Capitalize(op.ID)
 
-	// Don't add the input parameter argument unless there are some arguments
+	// Don't add the input parameter argument unless there are some arguments.
+	// If a method has a single input parameter, and it's a schema, make the
+	// generated type for that schema the input of the method.
+	// TODO: If a method has a single input parameter, and it's a simple type (int/string),
+	// make that the input of the method.
```

Comment:
@kvigen left this as a TODO since I think it will involve adding a new method to the test swagger definition, and so might be a bigger change involving new tests, etc.